### ISSUE TITLE
Small area flow compensation not working - RC and Stable

### ISSUE THREAD

**Issue by adolfotregosa** [SoftFever/OrcaSlicer#5780]
### Is there an existing issue for this problem?

- [X] I have searched the existing issues

### OrcaSlicer Version

2.1-rc and Stable

### Operating System (OS)

Linux

### OS Version

manjaro

### Printer

voron 2.4

### How to reproduce

[small area flow compensation test.zip](https://github.com/user-attachments/files/15919077/small.area.flow.compensation.test.zip)
Rotate Z 270º 

### Actual results

![image](https://github.com/SoftFever/OrcaSlicer/assets/7437931/7b38fead-38b7-4ee8-8782-b0e776f89cbe)

2.1 - rc

### Expected results

![image](https://github.com/SoftFever/OrcaSlicer/assets/7437931/e0372d11-a54c-4ae8-8364-69a7703da492)

2.1 beta

### Project file & Debug log uploads

[small area flow compensation test.zip](https://github.com/user-attachments/files/15919113/small.area.flow.compensation.test.zip)

### Checklist of files to include

- [ ] Log file
- [ ] Project file

### Anything else?

not working on RC

**Comment by adolfotregosa** [comment]
Not working on stable 2.1

Am I missing something ???

![image](https://github.com/SoftFever/OrcaSlicer/assets/7437931/32384357-ddf5-4e64-ba67-242ec71a77a9)

**Comment by drewmoseley** [comment]
Same behavior here.  I tested on the 2.1.0 release on both Ubuntu (AppImage) and Mac (arm64/DMG).

**Comment by mfish38** [comment]
Not working for me either. 2.1.0 release on Kubuntu AppImage

Screenshot taken with small area flow compensation on.
![image](https://github.com/SoftFever/OrcaSlicer/assets/20893814/9adc37f9-8697-41cf-9459-9f4dd3295c63)

**Comment by mjonuschat** [comment]
Looks like it was broken by https://github.com/SoftFever/OrcaSlicer/commit/9a4429f6037bb87001b78d9fe9f158bd5f1f4a61

**Comment by AlexDF99** [comment]
I have the same problem, (2.1.0 and 2.0.0 final release) on Windows 11

**Comment by TheSlashEffect** [comment]
The issue persists on Linux as well.
In its current state, `m_config.small_area_infill_flow_compensation.value` is not read as true when the feature is enabled, making the feature not activate.
If I revert https://github.com/SoftFever/OrcaSlicer/commit/9a4429f6037bb87001b78d9fe9f158bd5f1f4a61 the feature works as expected again.
Alternatively, if I move the definition of the config value small_area_infill_flow_compensation from PrintRegionConfig to PrintConfig, the feature works again.
The value was moved to the region config due to [this conversation](https://github.com/SoftFever/OrcaSlicer/pull/3334/commits/44665b39e1a43c29fa2bf9a4d566c89838545969#r1459245343):
I tried making the feature work again in its current state but to no avail. I ported the feature with the same PrintConfig setup on PrusaSlicer and SuperSlicer and it works as expected in both cases.
